### DISCUSSION TITLE
Why not align the node to 64 bytes?

### DISCUSSION THREAD

**Discussion by Bilibotter** [maypok86/otter#96] [General]
I've been reading the source code of Otter lately, and it's amazing.
I read the CLHT paper.In my understanding, it should align the node to 64 bytes.
But I found that it only aligns the bucket to 64 bytes, not the node.
There may be an error in my understanding, so I'm wondering about the reason for only aligning the bucket to 64 bytes.

**Comment by maypok86** [comment]
Hi, that's because when interacting with various nodes, there will be no [False Sharing](https://theboreddev.com/understanding-false-sharing/) (only pointers to them are stored in buckets). On the other hand, it is possible in buckets. For example, if you increase the number of nodes in each bucket.
<img width="706" alt="false-sharing" src="https://github.com/maypok86/otter/assets/72698179/95a2658d-605e-4b28-9d04-20e5f1aefaca">
